Documentation/btrfs-subvolume.rst

```diff
@@ -252,17 +252,23 @@ show [options] <path>
         -u|--uuid UUID
                 show details about subvolume with the given *UUID*, looked up in *path*
 
-snapshot [-r] [-i <qgroupid>] <source> <dest>|[<dest>/]<name>
+snapshot [-r|-R|--recursive] [-i <qgroupid>] <source> <dest>|[<dest>/]<name>
         Create a snapshot of the subvolume *source* with the
         name *name* in the *dest* directory.
 
         If only *dest* is given, the subvolume will be named the basename of *source*.
         If *source* is not a subvolume, btrfs returns an error.
 
+        If you wish to recursively create a readonly snapshot, you can run
+        :command:`btrfs property set <path> ro true` on each subvolume after this command completes.
```

Comment:
For clarity I'd suggest to mention that it has to be done from the leaves up in the tree consisting of subvolumes. Which may not be that easy to find out manually, so this could be a separate command for `subvolume` (in a separate patch). For parity an option to the snapshot command can be added that should read like it's doing the read-only change after the snapshots. This can be also added separatelly, we need to define the use cases first.